### PR TITLE
LVRUBYM-349:Sidebar component

### PR DESCRIPTION
**Pull request by HelenMaherovska**
## JIRA

* [Main JIRA ticket](http://ssu-jira.softserveinc.com/browse/LVRUBYM-344)
* [JIRA subtask](http://ssu-jira.softserveinc.com/browse/LVRUBYM-349)

## Code reviewers

- [x] @Mykhailo123 
- [x] @VitaliaDurnyak 
- [x] @andriyarterchuk

### Second Level Review

- [ ] @4asof4ik

## Summary of issue

Need to implement sidebar component

## Summary of change

- implemented sidebar component
- implemented styles

## Testing approach

Test with JSCS

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=90%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/js/main/components/main.template.html [hunk 1]
```diff
@@ -65,10 +47,46 @@
           <li><a ui-sref="#">Supplies</a></li>
           <li><a ui-sref="#">Our Day</a></li>
           <li><a ui-sref="#">My Day</a></li>
+=======
```
Comment (Mykhailo123) [inline app/js/main/components/main.template.html]: Delete

File: app/js/main/main.module.js [hunk 2]
```diff
@@ -5,7 +5,14 @@ var presenceReportModule = require('./../presenceReport/presenceReport.module.js
 var studentModule = require('./../student/student.module.js');
 var groupService = require('./../common/services/group.service.js');
 var staticModule = require('./../static/static.module.js');
+<<<<<<< HEAD
```
Comment (Mykhailo123) [inline app/js/main/main.module.js]: Same Here

File: app/js/static/static.route.js [hunk 3]
```diff
@@ -1,5 +1,6 @@
 module.exports = angular
   .module('static.route', [])
+<<<<<<< HEAD
```
Comment (Mykhailo123) [inline app/js/static/static.route.js]: Same Here

File: app/js/static/components/static.template.html [hunk 4]
```diff
@@ -10,8 +10,18 @@
     </div>
     <section class="student-btn" layout="column" layout-align="center center">
         <div class="btn-container" layout="row">
+<<<<<<< HEAD
```
Comment (Mykhailo123) [inline app/js/static/components/static.template.html]: Same Here

File: app/js/main/components/main.template.html [hunk 5]
```diff
@@ -65,10 +47,45 @@
...
+      <md-menu-item>
+        <md-button ng-click="$ctrl.logout()">Sign out</md-button>
+      </md-menu-item>
+    </md-menu-content>
+    </md-menu>
+    </md-button>
+</header>
+
+
+<div class="teacher-report" layout="row">
```
Comment (vikabelous) [inline app/js/main/components/main.template.html]: We still have this in main template

File: app/js/main/components/main.component.js [hunk 6]
```diff
@@ -16,21 +16,58 @@ MainController.$inject = [
...
+  'menuItems'
 ];
 
-function MainController($scope, $state, auth, currentGroupDay) {
+function MainController($scope, $state, auth, currentGroupDay, menuItems) {
   var ctrl = this;
 
   ctrl.currentGroupDay = currentGroupDay;
+  ctrl.menuItems = menuItems;
+  ctrl.activeMenu = localStorage.getItem('activeMenu') || 0;
```
Comment (vikabelous) [inline app/js/main/components/main.component.js]: Still in main component

File: app/js/sidebar/components/sidebar.component.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+SideBarController.$inject = [
+  'menuItems'
+];
+
+function SideBarController(menuItems) {
+  var ctrl = this;
+
+  ctrl.menuItems = menuItems;
+  ctrl.activeMenu = localStorage.getItem('activeMenu') || 0;
```
Comment (vikabelous) [inline app/js/sidebar/components/sidebar.component.js]: Why do you use local storage for this? Is it possible to store this in local ctrl variables?

Comment (HelenMaherovska) [inline app/js/sidebar/components/sidebar.component.js]: Local storage is used in order to store current value of menu even if the page was reloaded

File: app/js/main/components/main.component.js [hunk 8]
```diff
@@ -23,14 +23,27 @@ function MainController($scope, $state, auth, currentGroupDay) {
   var ctrl = this;
 
   ctrl.currentGroupDay = currentGroupDay;
+  ctrl.selectedDate = new Date();
```
Comment (vikabelous) [inline app/js/main/components/main.component.js]: This is related to calendar, it shouldn't be here also

File: app/js/main/components/main.template.html [hunk 9]
```diff
@@ -1,74 +1,48 @@
...
-          <span class="number">{{selectedDate.getDate() + 1}}</span>
-          <div class="date" layout="column">
-            <span class="month">{{monthNames[selectedDate.getMonth()] }}</span>
-            <span class="day">{{dayNames[selectedDate.getDay()] }}</span>
-          </div>
-        </div>
-        <a ui-sref="#" ng-click="nextDate()" ng-click="showMonth()" role="button">
-          <md-icon id="chevron_right" class="chevron">chevron_right</md-icon>
-        </a>
+    <div class="calendar" layout="row" layout-align="center start">
```
Comment (vikabelous) [inline app/js/main/components/main.template.html]: This shouldn't be here

Comment (HelenMaherovska) [inline app/js/main/components/main.template.html]: I have to do another component?

File: app/js/static/components/static.template.html [hunk 10]
```diff
@@ -21,6 +21,8 @@
     </md-list-item>
     </md-list>
 </div>
+
```
Comment (4asof4ik) [inline app/js/static/components/static.template.html]: Why you add 2 empty lines?

File: app/js/sidebar/sidebar.route.js [hunk 11]
```diff
@@ -0,0 +1,10 @@
+module.exports = angular
+  .module('sidebar.route', [])
+  .config(['$stateProvider', '$urlRouterProvider',
+    function($stateProvider, $urlRouterProvider) {
+    $urlRouterProvider.otherwise('/presence_report');
+    $stateProvider.state('main.sidebar', {
+      name: 'main.sidebar',
```
Comment (4asof4ik) [inline app/js/sidebar/sidebar.route.js]: 6 and 7 hmmm??

File: app/js/sidebar/components/sidebar.template.html [hunk 12]
```diff
@@ -0,0 +1,21 @@
...
+  <nav class="navbar">
+    <ul class="menu">
+      <div ng-repeat="menu in $ctrl.menuItems">
+        <li class="menu-item" ng-click="$ctrl.setActiveMenu($index)" ng-class="$ctrl.activeMenu == $index ? 'active-menu' : ''">
+          <a ui-sref={{menu.link}}>
+            {{menu.name}}
+          </a>
+        </li>
+        <ul class="submenu" ng-if="menu.submenu" ng-class="$ctrl.submenuOpened ? 'opened' : 'closed'">
+          <li ng-repeat="submenu in menu.submenu" ng-click="$ctrl.setActiveSubmenu($index)" ng-class="$ctrl.activeSubmenu == $index ? 'active-submenu' : ''">
```
Comment (4asof4ik) [inline app/js/sidebar/components/sidebar.template.html]: all logic should be in controller

File: app/js/static/components/static.template.html (hunk 10 above)
Comment (HelenMaherovska) [inline app/js/static/components/static.template.html]: I fixed that

File: app/js/sidebar/components/sidebar.template.html (hunk 12 above)
Comment (HelenMaherovska) [inline app/js/sidebar/components/sidebar.template.html]: I think the simpliest way to add active class it's use ng-class

File: index.html [hunk 13]
```diff
@@ -5,6 +5,8 @@
     <title>School area</title>
     <base href="/">
     <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
+    <link href="https://fonts.googleapis.com/css?family=Lato" rel="stylesheet">
+
```
Comment (Vovk-Nazar) [inline index.html]: Drop empty line

File: app/js/sidebar/components/sidebar.template.html [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+  <nav class="navbar">
+    <ul class="menu">
+      <div ng-repeat="menu in $ctrl.menuItems">
+        <li class="menu-item" ng-click="$ctrl.setActiveMenu($index)" ng-class="$ctrl.activeMenu == $index ? 'active-menu' : ''">
+          <a ui-sref={{menu.link}}>
+            {{menu.name}}
+          </a>
+        </li>
+        <ul class="submenu" ng-class="$ctrl.submenuOpened ? 'opened' : 'closed'">
+          <li ng-repeat="submenu in menu.submenu" ng-click="$ctrl.setActiveSubmenu($index)" ng-class="$ctrl.activeSubmenu == $index ? 'active-submenu' : ''">
```
Comment (Vovk-Nazar) [inline app/js/sidebar/components/sidebar.template.html]: 5 and 11  to much attr in one line ask Vita how fix this.
And check your templates

**Pull request merged**